ntoskrnl/include/internal/io.h

```diff
@@ -321,7 +321,7 @@ typedef struct _IO_WORKITEM
 typedef struct _IO_INTERRUPT
 {
     KINTERRUPT FirstInterrupt;
-    PKINTERRUPT Interrupt[MAXIMUM_PROCESSORS];
+    PKINTERRUPT Interrupt[MAXIMUM_PROCESSORS - 1];
```

Comment:
Wouldn't it be "simpler" to store the interrupt actual/allocated count, instead of that interrupt and pointers mix?